### PR TITLE
fix: publish events about unannounced applications

### PR DESCRIPTION
**Pull request by chriswk**
Tried to get the test working with fake clocks, but just ended up in a weird pending state, as such, this isn't optimal and I'm more than happy if someone can think of a better way of checking that we do announce the created applications, and just once.

### CODE REVIEW

File: src/migrations/20210304141005-add-announce-field-to-application.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+exports.up = function(db, cb) {
+    db.runSql(
+        `
+    ALTER TABLE client_applications ADD COLUMN announced boolean DEFAULT false;
```
Comment (ivarconr) [inline src/migrations/20210304141005-add-announce-field-to-application.js:6]: does this end up announcing all existing applications in our DB?

Comment (ivarconr) [reply src/migrations/20210304141005-add-announce-field-to-application.js:6]: > The new column is initially filled with whatever default value is given (null if you don't specify a DEFAULT clause).

https://www.postgresql.org/docs/9.5/ddl-alter.html

Comment (ivarconr) [reply src/migrations/20210304141005-add-announce-field-to-application.js:6]: could we just add a: 

`UPDATE client_applications set announced=true`

Comment (chriswk) [reply src/migrations/20210304141005-add-announce-field-to-application.js:6]: I was afraid of locking. So the query that looks for unannounced applications also filters out rows with the column set to `null`. That should avoid announcing all apps. 
I just know from earlier projects that updating all rows in a migration can be a bit risky, but we might not be speaking millions of rows here, so maybe premature worrying.

Comment (chriswk) [reply src/migrations/20210304141005-add-announce-field-to-application.js:6]: After checking the doc, I see we've already set all rows to 'f', I've added the extra
`Update client_applications set announced=true` query

**Pull request merged**